heightNow req_height
---
uint8 finished